lib/utils/patterns.ts

```diff
@@ -282,6 +283,69 @@ export function inferOrSelectDevice(preferredUuid: string) {
 		});
 }
 
+export function getTargetUuid(
+	sdk: BalenaSdk.BalenaSDK,
+	applicationOrDevice: string,
+): Bluebird<string> {
+	const appTest = validation.validateApplicationName(applicationOrDevice);
+	const uuidTest = validation.validateUuid(applicationOrDevice);
+
+	if (!appTest && !uuidTest) {
+		throw new Error(`Device or application not found: ${applicationOrDevice}`);
+	}
+
+	const isBalenaApplicationNotFound = (e: any) =>
+		e instanceof BalenaApplicationNotFound;
+
+	// if we have a definate device UUID...
+	if (uuidTest && !appTest) {
+		return sdk.models.device
+			.get(applicationOrDevice, { $select: ['uuid'] })
+			.then(d => d.uuid);
+	}
+
+	// otherwise, it may be a device OR and application...
+	return sdk.models.application
+		.get(applicationOrDevice)
+		.then(app => {
+			return sdk.models.device.getAllByApplication(app.id, {
+				$filter: { is_online: true },
+			});
+		})
+		.then(devices => {
+			if (_.isEmpty(devices)) {
+				throw new Error('No target devices are online');
+			}
+
+			return getForm().ask({
+				message: 'Select a device',
+				type: 'list',
+				default: devices[0].uuid,
+				choices: _.map(devices, device => ({
+					name: `${device.device_name || 'Untitled'} (${device.uuid.slice(
+						0,
+						7,
+					)})`,
+					value: device.uuid,
+				})),
+			}) as string;
```

Comment:
This cast is slightly weird, it's actually a promise that this function returns.

Reply:
If you used an `await` this would happily resolve itself.